src/data_analysis/symmetry_checker.py

```diff
@@ -0,0 +1,239 @@
+import re
+import sys
+import warnings
+from pathlib import Path
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+from patient.patient_data_loader import PatientDataLoader
+from utils.everion_keys import EverionKeys
+
+
+def ensure_dir(path):
+    path = Path(path)
+    if not path:
+        return False
+    if not path.is_dir():
+        path.mkdir(parents=True, exist_ok=True)
+    return path.is_dir()
+
+
+def save_figure(path, fig=None, dpi=300):
+    path = Path(path)
+    if fig is not None:
+        # Make the figure with fig the current figure
+        plt.figure(fig.number)
+    if not ensure_dir(path.parent):
+        assert False, "Failed to create output directory: %s " % path.parent
+    plt.savefig(path, bbox_inches="tight", dpi=dpi)
+
+
+class SymmetryChecker:
+    """
+    Patients wear a sensor on the left and right side. This utility analyzes
+    and visualizes the relationship between these data sources. Most notably
+    it estimates (per-patient) the Pearson correlation coefficient between
+    the left and right signals, and creates Bland-Altman plots for a graphical
+    analysis.
+    """
+
+    def __init__(self, data_dir, out_dir, columns, resample=None):
+        """
+        Arguments:
+            data_dir: Input folder with the .csv files. The following naming
+                      convention applies for the .csv files:
+                            ([0-9]*)(L|R).*
+                      The first capture group identifies the patient id, the
+                      second one whether the data is from left or right side.
+                            001L_storage-vital.csv
+                            001R_storage-vital.csv
+            out_dir: Path to output directory
+            columns: Columns to select ("columns of interest", coi)
+            resample: Optionally resample the data. For instance, setting
+                      resample="30s" aggregates the data into 30s bins.
+                      See doc of pd.DataFrame.resample for details.
+
+        """
+        self._loader = PatientDataLoader()
+        self._data_dir = Path(data_dir)
+        self._out_dir = Path(out_dir)
+        self._resample = resample
+        # Columns of interest.
```

Comment:
I try to use as little comments as possible since they often become obsolete in the future. in this situation, I would use a variable name that is self-explainable. e.g., columns

Reply:
I agree!